src/test/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManagerTest.java

```diff
@@ -89,535 +105,1362 @@ public final void setUp() {
+      dummyEventBookingPersistenceManager.lockEventUntilTransactionComplete(dummyTransaction, testCase.event.getId());
+      expectLastCall().once();
+
+      // Check existing bookings
+      expect(dummyEventBookingPersistenceManager.getBookingsByEventId(testCase.event.getId()))
+          .andReturn(ImmutableList.of(student2sCancelledReservation)).once();
+      expect(dummyEventBookingPersistenceManager
+          .getEventBookingStatusCounts(testCase.event.getId(), false))
+          .andReturn(previousBookingCounts).once();
+
+      // Make Reservations
+      expect(dummyTransactionManager.getTransaction()).andReturn(dummyTransaction).once();
+
+      expect(dummyEventBookingPersistenceManager
+          .getBookingByEventIdAndUserId(testCase.event.getId(), testCase.student1.getId()))
+          .andReturn(null).once();
+      expect(dummyEventBookingPersistenceManager
+          .createBooking(eq(dummyTransaction), eq(testCase.event.getId()), eq(testCase.student1.getId()),
+              eq(testCase.teacher.getId()), eq(BookingStatus.RESERVED), anyObject()))
+          .andReturn(testCase.student1Booking).once();
+
+      dummyTransaction.commit();
+      expectLastCall().once();
+      dummyTransaction.close();
+      expectLastCall().once();
+
+      // Send Emails
+      expect(dummyEmailManager.getEmailTemplateDTO(("email-event-reservation-requested"))).andReturn(
+          testCase.reservationEmail).atLeastOnce();
+      expect(dummyEmailManager.getEmailTemplateDTO(("email-event-reservation-recap"))).andReturn(
+          testCase.reservationEmail).atLeastOnce();
+
+      expect(dummyUserAccountManager.getUserDTOById(testCase.student1.getId())).andReturn(testCase.student1)
+          .atLeastOnce();
+      dummyEmailManager.sendTemplatedEmailToUser(eq(testCase.student1), eq(testCase.reservationEmail), anyObject(),
+          eq(EmailType.SYSTEM));
+      expectLastCall().once();
+      dummyEmailManager.sendTemplatedEmailToUser(eq(testCase.teacher), eq(testCase.reservationEmail), anyObject(),
+          eq(EmailType.SYSTEM));
+      expectLastCall().atLeastOnce();
+
+      // Run the test for a student event
+      replay(mockedObjects);
+      List<EventBookingDTO> actualResults =
+          eventBookingManager.requestReservations(testCase.event, students, testCase.teacher);
+      List<EventBookingDTO> expectedResults = ImmutableList.of(testCase.student1Booking);
+      assertEquals(expectedResults, actualResults,
+          "Student 1 should get reserved despite the existing cancelled reservation");
+      verify(mockedObjects);
+    }
   }
 
-  @Test
-  void getEventPage_checkStudentEventReservedBookings_capacityCalculatedCorrectly() throws
-      Exception {
-    EventBookingManager ebm = this.buildEventBookingManager();
-    IsaacEventPageDTO testEvent = prepareIsaacEventPageDto(studentCSTags, 2, EventStatus.OPEN);
-    testEvent.setAllowGroupReservations(true);
+  @Nested
+  class CancelBooking {
```

Comment:
Lots and lots of commented out code. Also, this test class is HUGE!

Reply:
Oh, I should probably take out the cancel tests properly...
I pulled over some updated/expanded tests from one of the other tickets in review to ensure things were covered but that section doesn't work without those changes...